api/turing/api/deployment_controller.go

```diff
@@ -188,14 +188,23 @@ func (c RouterDeploymentController) deployRouterVersion(
 		}
 	}
 
+	expSvc := c.BaseController.AppContext.ExperimentsService
 	if routerVersion.ExperimentEngine.Type != models.ExperimentEngineTypeNop {
-		// If passkey has been set, decrypt it
-		if routerVersion.ExperimentEngine.Config.Client.Passkey != "" {
-			experimentPasskey, err = c.CryptoService.Decrypt(routerVersion.ExperimentEngine.Config.Client.Passkey)
+		if expSvc.IsStandardExperimentManager(string(routerVersion.ExperimentEngine.Type)) {
```

Comment:
Why do we need to cast it to string here?

Reply:
The experiment service methods have always worked with strings. Eg: https://github.com/gojek/turing/blob/main/api/turing/service/experiment_service.go#L22

But in the RouterVersion model, this value is of type [models.ExperimentEngineType](https://github.com/gojek/turing/blob/main/api/turing/models/experiment_engine.go#L14). There is some inconsistency with how the value is processed but I am also considering removing this custom type altogether later on, as I feel there is little use for it (or it could be defined in the `experiment/engines` package, with the possible values being dynamically generated based on what engines are registered).